content/blog/2024-05-23-linebender-roadmap.md

```diff
@@ -0,0 +1,248 @@
+
+### Add basic benchmarks
+
+- **Priority:** Low.
+- **Assigned:** Olivier.
+- **Description:** Right now we're mostly working off vibes to know whether or not any of our work improved performance or introduced regressions.
+We should write at least *one* benchmark that would test a basic Xilem use-case, so we get a general idea how our performance is progressing.
+- **Issue:** [xilem#362](https://github.com/linebender/xilem/issues/362)
+
+
+## AccessKit
+
+Matt Campbell's contract with Google hasn't started yet.
+When it does, top priority will likely be the Android backend.
+
+The plan is to study existing Android accessibility implementations for non-Java toolkits, including Flutter, Chromium, and Gio, to get a better understanding of what's required to implement Android accessibility from scratch.
+
+Beyond Android support, we've only had cursory discussions of what work Matt will do for the rest of the year.
+Possible themes include scrollable containers, virtualized lists, richer text support (e.g. exposing font and style attributes) across platforms, and better integration into Xilem/Masonry.
+
+
+## Android-specific work
+
+Getting the Android port right involves a lot of small changes and making things feel non-broken.
+
+Some notable sub-goals:
+
+### Subclassing View/Application/Activity
+
+- **Priority:** Highest.
+- **Assigned:** Aaron Muir Hamilton, Raph Levien.
+- **Description:** The Java classes of an Android application are where the rubber meets the road. We're investigating what we want our boilerplate to look like.
+
+### Handling IME
+
+- **Priority:** Highest.
+- **Assigned:** Aaron Muir Hamilton, Raph Levien.
+- **Description:** We need to extend Winit to be compatible with Android IME interfaces. An Android application without IME is a non-starter. Our design choices here will have influence on accessibility as well.
+
+### Gesture recognition
+
+- **Priority:** Low.
+- **Assigned:** Aaron Muir Hamilton.
+- **Description:** When the user swipes on the screen, we want to send some kind of scrolling events instead of a normal pointer event. Ideally there should be first-class support for that event to handle momentum scrolling, but a dumber solution would just emulate mouse wheel ticks.
+
+
+## Conclusion
+
+Whew! We have our work cut out for us there.
+
+Overall, I'm feeling very optimistic about the coming months.
+In the weeks since we first drafted this roadmap, we've already made a lot of progress on our respective tasks (for instance Daniel is now running on what async experts call a "work-stealing" algorithm), so it does look like we'll meet our deadlines.
+
+What comes after that is a bit hazier.
+Our three areas of focus are likely to be Android support, accessibility, and developer experience.
+
+As we fill out gaps in our feature set, Xilem is moving closer to becoming a stable, mature framework for writing GUI applications.
+We're not GUI yet, but we hope we'll be in the near future.
```

Comment:
This is potentially "promising" more than we're willing to give (especially considering Raph's question answer in the RustNL talk). I like the optimistic sentiment.

I'd maybe rephrase it in terms of "filling out the features needed for Xilem to thrive"?

Reply:
I've changed the wording a bit to make it more of a "north star" thing.